docs/man/sbaz.1
.\" ##########################################################################
.\" #                      __                                                #
.\" #      ________ ___   / /  ___     Scala 2 On-line Manual Pages          #
.\" #     / __/ __// _ | / /  / _ |    (c) 2002-2007, LAMP/EPFL              #
.\" #   __\ \/ /__/ __ |/ /__/ __ |                                          #
.\" #  /____/\___/_/ |_/____/_/ | |    http://scala-lang.org/                #
.\" #                           |/                                           #
.\" ##########################################################################
.\"
.\" Process this file with nroff -man scala.1
.\"
.TH sbaz 1  "August 24, 2006" "version 0.3" "USER COMMANDS"
.\"
.\" ############################## NAME ###############################
.\"
.SH NAME
.PP
\fBsbaz\fR \- Scala package sharing tool for the Scala 2 language
.\"
.\" ############################## SYNOPSIS ###############################
.\"
.SH SYNOPSIS
.PP
\fBsbaz\fR  [ <\fIglobal_options\fR> ] <\fIcommand\fR> [ <\fIcommand_options\fR> ]
.\"
.\" ############################## PARAMETERS ###############################
.\"
.SH PARAMETERS
.PP
.TP
<\fIglobal_options\fR>
Command line options. See \fBOPTIONS\fR below.
.TP
<\fIcommand\fR>
Internal \fBsbaz\fR command.
.TP
<\fIcommand_options\fR>
\fBsbaz\fR command options.
.\"
.\" ############################## DESCRIPTION ###############################
.\"
.SH DESCRIPTION
.PP
The \fBsbaz\fR tool is a system used by Scala enthusiasts to share computer files with each other. In particular, it makes it easy to share libraries and applications.
.\"
.\" ############################## OPTIONS ###############################
.\"
.SH OPTIONS
.PP
The \fBsbaz\fR tool has a set of standard options that are supported on the current development environment and will be supported in future releases.
.\"
.\" ############################## Global Options ###############################
.\"
.SS "Global Options"
.PP
.TP
\fB\-d\fR <\fIdir\fR> 
Operate on dir as the local managed directory.
.TP
\fB\-n\fR | \fB\-\-dryrun\fR 
Do not actually do anything.  Only print out what tool would normally do with the following arguments.
.TP
\fB\-v\fR | \fB\-\-verbose\fR 
Output messages about what the \fBsbaz\fR tool is doing
.TP
\fB\-version\fR 
Display the version information
.TP
\fB\--univ\fR <\fIname\fR>
Operate on the named remote universe, selected from those in the local managed directory's universe. Affects the \fBshare\fR and \fBretract\fR commands.
.TP
\fB\--univ-url\fR <\fIurl\fR>
Operate on the universe at the specified URL.  Affects the \fBshare\fR and \fBretract\fR commands.
.\"
.\" ############################## Available Commands ###############################
.\"
.SS "Available Commands"
.PP
.TP
\fBavailable\fR
List the available packages for installation; only display the three most recent versions of each package.
.TP
\fBavailable\fR \fB\-a\fR 
List the available packages for installation; display all shared versions of each package.
.TP
\fBcompact\fR
Clear the download cache to save space.
.TP
\fBhelp\fR
Display a help message.
.TP
\fBinstall\fR
Install a package.
.TP
\fBinstalled\fR
List the packages that are installed.
.TP
\fBkeycreate\fR
Request that a new key be created.
.TP
\fBkeyforget\fR
Forget the specified key.
.TP
\fBkeyknown\fR
List all known keys.
.TP
\fBkeyremember\fR
Remember the specified key for future use.
.TP
\fBkeyremoteknown\fR
List all keys known to the bazaar server.
.TP
\fBkeyrevoke\fR
Request that a specified key be revoked.
.TP
\fBpack\fR <\fIname\fR> <\fIdirectory\fR> [ <\fIoptions\fR> ]
Create an sbaz package and, if a link base is specified, an advertisement file. The package file is named name-version.sbp. The advertisement file is named name-version.advert. The URL is the advertisement file is the URL base with the package filename appended.
.TP
\fBremove\fR
Remove a package.
.TP
\fBretract\fR
Retract a previously shared package.
.TP
\fBsetuniverse\fR
Set the universe for a directory.
.TP
\fBsetup\fR
Initialize a directory to be managed.
.TP
\fBshare\fR <\fIfilename\fR>
Share a package advertisement on a bazaar.
.TP
\fBshare\fR \fB\-i\fR <\fIdescriptor\fR> 
The package advertisement is usually specified in a file, but it may also be specified on the command line with the \fB\-i\fR  option.
.TP
\fBshare\fR \fB\-\-template\fR 
If \fB\-\-template\fR  is specified, then instead of uploading a description, the command prints out a template of a package advertisement.
.TP
\fBshow\fR
Show information about one package.
.TP
\fBupdate\fR
Update the list of available packages.
.TP
\fBupgrade\fR
Upgrade all possible packages.
.\"
.\" ############################## EXAMPLES ###############################
.\"
.SH EXAMPLES
.PP
.TP
Update the list of available packages.
\fBsbaz\fR \fBupdate\fR
.TP
Upload package description for scala-devel-2.5.1 to the universe
\fBsbaz\fR \fBshare\fR scala-devel-2.5.1.advert
.\"
.\" ############################## EXIT STATUS ###############################
.\"
.SH "EXIT STATUS"
.PP
\fBsbaz\fR returns a zero exist status if it succeeds to process the specified input files. Non zero is returned in case of failure.
.\"
.\" ############################## AUTHOR ###############################
.\"
.SH AUTHOR
.PP
Written by Lex Spoon.
.\"
.\" ############################## REPORTING BUGS ###############################
.\"
.SH "REPORTING BUGS"
.PP
Report bugs to <scala@listes.epfl.ch>.
.\"
.\" ############################## COPYRIGHT ###############################
.\"
.SH COPYRIGHT
.PP
This is open-source software, available to you under a BSD-like license. See accomponying "copyright" or "LICENSE" file for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
.\"
.\" ############################## SEE ALSO ###############################
.\"
.SH "SEE ALSO"
.PP
\fBfsc\fR(1), \fBscala\fR(1), \fBscalac\fR(1), \fBscaladoc\fR(1), \fBscalap\fR(1)


debian/changelog
scala (2.6.1-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Tue, 15 Jan 2008 18:08:31 +0100

scala (2.6.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Fri, 14 Sep 2007 15:22:05 +0100

scala (2.5.0-1) unstable; urgency=low

  * New upstream release
  * Depend only on java-gcj-compat, not java-gcj-compat-dev

 -- Lex Spoon <lex@debian.org>  Tue, 22 May 2007 16:51:47 +0200

scala (2.4.0-1) unstable; urgency=low

  * New upstream version.

  * Deleting all mention of the Sun JVM, just because some
    DD's find it bothersome.

  * Deleted mention of JDK5-specific classes, while awaiting
    the Sun JVM to become open source.  (Closes: #409785)

  * Added java-gcj-compat-dev as an install dependency.  (Closes: #408353)

 -- Lex Spoon <lex@debian.org>  Sun, 25 Mar 2007 11:56:18 +0200

scala (2.3.2-1) unstable; urgency=low

  * New upstream version

 -- Lex Spoon <lex@debian.org>  Fri, 19 Jan 2007 14:45:51 +0100

scala (2.3.0-1) unstable; urgency=low

  * New upstream version
  * Added a Java compiler as a build-dependency

 -- Lex Spoon <lex@debian.org>  Fri, 24 Nov 2006 16:02:09 +0100

scala (2.2.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Mon, 30 Oct 2006 21:35:10 -0500

scala (2.1.5-2) unstable; urgency=low

  * Rearranged to have three packages instead of four.

 -- Lex Spoon <lex@debian.org>  Wed, 28 Jun 2006 17:16:02 +0200

scala (2.1.5-1) unstable; urgency=low

  * upstream update.

 -- Lex Spoon <lex@debian.org>  Sun, 9 Jun 2006 14:45:26 +0100

scala (2.1.1-1) unstable; urgency=low

  * Initial Release.

 -- Lex Spoon <lex@debian.org>  Sun, 26 Mar 2006 14:45:26 +0100



src/manual/scala/man1/sbaz.scala
/* NSC -- new Scala compiler
 * Copyright 2005-2007 LAMP/EPFL
 * @author Stephane Micheloud
 */
//$Id$

package scala.man1

object sbaz extends Command {
  import _root_.scala.tools.docutil.ManPage._

  protected val cn = new Error().getStackTrace()(0).getClassName()

  val name = Section("NAME",

    MBold(command) & " " & NDash & " Scala package sharing tool for the " &
    Link("Scala 2", "http://scala-lang.org/") & " language")

  val synopsis = Section("SYNOPSIS",

    CmdLine(" [ " & Argument("global_options") & " ] " & Argument("command") &
            " [ " & Argument("command_options") & " ]"))

  val parameters = Section("PARAMETERS",

    DefinitionList(
      Definition(
        Mono(Argument("global_options")),
        "Command line options. See " & Link(Bold("OPTIONS"), "#options") &
        " below."),
      Definition(
        Mono(Argument("command")),
        "Internal " & MBold(command) & " command."),
      Definition(
        Mono(Argument("command_options")),
        MBold(command) & " command options.")))

  val description = Section("DESCRIPTION",

    "The " & MBold(command) & " tool is a system used by Scala enthusiasts " &
    "to share computer files with each other. In particular, it makes it " &
    "easy to share libraries and applications.")

  val options = Section("OPTIONS",

    "The " & MBold(command) & " tool has a set of standard options that are " &
    "supported on the current development environment and will be supported " &
    "in future releases.",

    Section("Global Options",
      DefinitionList(
        Definition(
          CmdOption("d", Argument("dir")),
          "Operate on dir as the local managed directory."),
        Definition(
          CmdOption("n") & "| " & CmdOptionLong("dryrun"),
          "Do not actually do anything.  Only print out what " +
          "tool would normally do with the following arguments."),
        Definition(
          CmdOption("v") & "| " & CmdOptionLong("verbose"),
          "Output messages about what the " & MBold(command) & " tool is doing"),
        Definition(
          CmdOption("version"),
          "Display the version information"),

        Definition(
          CmdOption("-univ") & Argument("name"),
          "Operate on the named remote universe, selected from those " &
          "in the local managed directory's universe. Affects "&
          "the "&MBold("share")&" and "&MBold("retract")&" commands."),

        Definition(
          CmdOption("-univ-url") & Argument("url"),
          "Operate on the universe at the specified URL.  Affects "&
          "the "&MBold("share")&" and "&MBold("retract")&" commands."))),

    Section("Available Commands",
      DefinitionList(
        Definition(
          MBold("available"),
          "List the available packages for installation; only display the " +
          "three most recent versions of each package."),
        Definition(
          MBold("available") & " " & CmdOption("a"),
          "List the available packages for installation; display all shared " +
          "versions of each package."),
        Definition(
          MBold("compact"),
          "Clear the download cache to save space."),
        Definition(
          MBold("help"),
          "Display a help message."),
        Definition(
          MBold("install"),
          "Install a package."),
        Definition(
          MBold("installed"),
          "List the packages that are installed."),
        Definition(
          MBold("keycreate"),
          "Request that a new key be created."),
        Definition(
          MBold("keyforget"),
          "Forget the specified key."),
        Definition(
          MBold("keyknown"),
          "List all known keys."),
        Definition(
          MBold("keyremember"),
          "Remember the specified key for future use."),
        Definition(
          MBold("keyremoteknown"),
          "List all keys known to the bazaar server."),
        Definition(
          MBold("keyrevoke"),
          "Request that a specified key be revoked."),
        Definition(
          MBold("pack") & " " & Argument("name") & " " & Argument("directory") &
          " [ " & Argument("options") & " ]",
          "Create an sbaz package and, if a link base is specified, "&
          "an advertisement file. The package file is named " &
          Mono("name-version.sbp") & ". The advertisement file is named " &
          Mono("name-version.advert") & ". The URL is the advertisement " &
          "file is the URL base with the package filename appended."),
        Definition(
          MBold("remove"),
          "Remove a package."),
        Definition(
          MBold("retract"),
          "Retract a previously shared package."),
        Definition(
          MBold("setuniverse"),
          "Set the universe for a directory."),
        Definition(
          MBold("setup"),
          "Initialize a directory to be managed."),
        Definition(
          MBold("share") & " " & Argument("filename"),
          "Share a package advertisement on a bazaar."),
        Definition(
          MBold("share") & " " & CmdOption("i", Argument("descriptor")),
          "The package advertisement is usually specified in a file, " &
          "but it may also be specified on the command line with the " &
          CmdOption("i") & " option."),
        Definition(
          MBold("share") & " " & CmdOptionLong("template"),
          "If " & CmdOptionLong("template") & " is specified, then instead " &
          "of uploading a description, the command prints out a template " &
          "of a package advertisement."),
        Definition(
          MBold("show"),
          "Show information about one package."),
        Definition(
          MBold("update"),
          "Update the list of available packages."),
        Definition(
          MBold("upgrade"),
          "Upgrade all possible packages."))))

  val examples = Section("EXAMPLES",

    DefinitionList(
      Definition(
        "Update the list of available packages.",
        CmdLine(MBold("update"))),
      Definition(
        "Upload package description for " & Mono("scala-devel-2.5.1") &
        " to the universe",
        CmdLine(MBold("share") & " scala-devel-2.5.1.advert"))))

  val exitStatus = Section("EXIT STATUS",

    MBold(command) & " returns a zero exist status if it succeeds to process " &
    "the specified input files. Non zero is returned in case of failure.")

  override val authors = Section("AUTHOR",

    "Written by Lex Spoon.")

  val seeAlso = Section("SEE ALSO",

    Link(Bold("fsc") & "(1)", "fsc.html") & ", " &
    Link(Bold("scala") & "(1)", "scala.html") & ", " &
    Link(Bold("scalac") & "(1)", "scalac.html") & ", " &
    Link(Bold("scaladoc") & "(1)", "scaladoc.html") & ", " &
    Link(Bold("scalap") & "(1)", "scalap.html"))

  def manpage = new Document {
    title = command
    date = "August 24, 2006"
    author = "Stephane Micheloud"
    version = "0.3"
    sections = List(
      name,
      synopsis,
      parameters,
      description,
      options,
      examples,
      exitStatus,
      authors,
      bugs,
      copyright,
      seeAlso)
  }
}
